File: ShellingLaptopApp/app/src/main/java/com/shellinglaptop/db/CartDao.java
package com.shellinglaptop.db;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.shellinglaptop.model.Cart;
import com.shellinglaptop.model.Laptop;

import java.util.List;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface CartDao {

    @Query("SELECT * FROM cart")
    List<Cart> getCart();

    @Insert(onConflict = REPLACE)
    void insertCart(Cart cart);

    @Delete
    void deleteCart(Cart cart);
}

File: ShellingLaptopApp/app/src/main/java/com/example/shellinglaptopapp/data/offline/CartDao.kt
package com.example.shellinglaptopapp.data.offline

import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.shellinglaptopapp.data.model.Cart

@Dao
interface CartDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(cart: Cart)

    @Query("SELECT * FROM table_cart")
    fun getAllCarts(): LiveData<List<Cart>>

    @Delete
    suspend fun delete(cart: Cart)
}